src/main/webapp/database-tables.js

```diff
@@ -80,12 +80,55 @@ function createTable(name) {
 }
 
 function mainLoad(){
-  showDatabase();
+  console.log("main");
   login();
+  showReason();
 }
 
 function login() {
   fetch("/login").then(response => response.json()).then((user) => {
     document.getElementById("user").innerText = user;
+    var currentUser = user;
+    if (currentUser[0] == "Stranger"){
+        window.location.assign("https://accounts.google.com/signin/v2/identifier?");
+        return;
+    }
+
+    var usersWithAccess = ["jiaxinz@google.com","gagomez@google.com",/*"test@example.com",*/"hilakey@google.com","sasymer@google.com","williamdc@google.com"];
+    console.log(currentUser[0]);
+    console.log("current user");
+    for (var i =0; i < usersWithAccess.length; i++){
+        if (currentUser[0] == usersWithAccess[i]){
+            return;
+        }
+    }
```

Comment:
Let's move this check to the server and return whether the user has permission or not. Alternatively we can also redirect from the response to /denied.html and to the login URL in the case the user is not logged in.

Reply:
Most of this logic has been moved to the server but the redirects stayed in JS